src/lib/TurboConfirm.js

```diff
@@ -0,0 +1,104 @@
+import ConfirmationController from './ConfirmationController'
+
+export class TurboConfirm {
+  #controller
+  #config = {
+    dialogSelector: '#confirm',
+    activeClass: 'modal--active',
+    acceptSelector: '#confirm-accept',
+    denySelector: '.confirm-cancel',
+    animationDuration: 300,
+    showConfirmCallback: null,
+    messageSlotSelector: '#confirm-title',
+    contentSlots: {
+      body: {
+        contentAttribute: 'confirm-details',
+        slotSelector: '#confirm-body'
+      },
+      acceptText: {
+        contentAttribute: 'confirm-button',
+        slotSelector: '#confirm-accept'
+      }
+    }
+  }
+
+  constructor(options = {}) {
+    for(const [key, value] of Object.entries(options)) {
+      this.#config[key] = value
+    }
+
+    this.#controller = new ConfirmationController(this.#config.dialogSelector, this)
+  }
+
+  confirm(message, _formElement, submitter) {
+    const clickTarget = this.#clickTarget(submitter)
+    const contentMap = this.#contentMap(message, clickTarget)
+
+    return this.confirmWithContent(contentMap)
+  }
+
+  confirmWithContent(contentMap) {
+    return new Promise(resolve => this.#controller.showConfirm(contentMap, resolve))
+  }
+
+  showConfirm(element) {
+    element.classList.add(this.#config.activeClass)
+    // support dialog element
+    if (typeof element.showModal === 'function') {
+      element.showModal()
+    }
+    // is this still required?
+    if (this.#config.showConfirmCallback) {
+      this.#config.showConfirmCallback(element)
+    }
+  }
+
+  hideConfirm(element) {
+    element.classList.remove(this.#config.activeClass)
+    // support dialog element
+    if (typeof element.close === 'function') {
+      element.close()
+    }
```

Comment:
considering adding a `hideConfirmCallback` config option and making this the default value.